processing/data_collection/gazette/spiders/sp_guaruja.py

```diff
@@ -8,30 +10,88 @@
 
 
 class SpGuaruja(BaseGazetteSpider):
+
     TERRITORY_ID = "3518701"
     name = "sp_guaruja"
     allowed_domains = ["guaruja.sp.gov.br"]
-    start_urls = ["http://www.guaruja.sp.gov.br/index.php/diario-oficial/"]
+    start_urls = ["http://www.guaruja.sp.gov.br/edicoes-diario-oficial/"]
+
+    def __init__(self, start_date=None, end_date=None, *args, **kwargs):
+        self.start_date = dt.date(2009, 1, 1)
+        self.end_date = dt.date.today()
+
+        super(SpGuaruja, self).__init__(start_date, end_date, *args, **kwargs)
+
+        self.logger.debug(
+            "Start date is {date}".format(date=self.start_date.isoformat())
+        )
+        self.logger.debug("End date is {date}".format(date=self.end_date.isoformat()))
```

Comment:
If the goal of the __init__ is just initialize the start_date and end_date, we can do as the other member variables (name, start_urls, etc) in the class definition. Just to keep consistent with the other variables.

Furthermore, if the debug messages are important, we should move them to the base class. Thus, all the spider will print that info.

Reply:
The base class does not offer a default value for those dates, so it's impossible to log correctly from there.

Reply:
You can move them to the base class. 

1. define default values when declaring class attributes
2. on the base class, override the `__init__` method and log those values

I think it covers most cases:

- no default value on class / no value defined by instance
- no default value on class / value defined by instance
- default value on class / no value defined by instance
- default value on class / value defined by instance

What do you think?